pkg/edvprovider/edvprovider.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package edvprovider
+
+import (
+	"errors"
+
+	"github.com/trustbloc/edv/pkg/restapi/edv/models"
+)
+
+// ErrIndexingNotSupported is returned when an attempt is made to create an index in a provider that doesn't support it.
+var ErrIndexingNotSupported = errors.New("indexing is not supported by this provider")
+
+// EDVProvider represents a provider with functionality needed for EDV data storage.
```

Comment:
In the implementations of EDVProvider, I wrap an edge-core provider. I do this to abstract out EDV-specific operations that don't belong in edge-core.